engines/registry/app/models/registry/authenticated_client.rb

```diff
@@ -14,12 +14,19 @@ def initialize(login, password)
 
   private
 
-  def authenticate_by_system_credentials(login, password)
+  def authenticate_by_system_credentials(login, password, remote_ip)
     @systems = System.get_by_credentials(login, password)
     expiration_value = Settings[:registry].try(:token_expiration) || 8.hours.to_i
-    if @systems.any? { |system| system.last_seen_at > expiration_value.seconds.ago }
-      @account = login
-      @auth_strategy = :system_credentials
+    cache_key = [remote_ip, login].join('-')
+    registry_cache_path = Rails.root.join('tmp', 'registry', 'cache', cache_key)
+    cache_created = File.exist?(registry_cache_path)
+    if !@systems.empty? && cache_created
+      is_registry_cache_active = File.ctime(registry_cache_path) > expiration_value.seconds.ago
+      if is_registry_cache_active
+        @account = login
+        @auth_strategy = :system_credentials
+      end
+      File.delete(registry_cache_path) unless is_registry_cache_active
```

Comment:
I think that line is not needed, because you'll need a cronjob anyway(?) to wipe out cache files from that directory older 8h.

Reply:
Yes, in case a user do not come back for a period of time, those files would remain on disk
so probably it is better to have a task to remove those altogether instead of when accessing a single one